index.js

```diff
@@ -4,79 +4,114 @@ import cheerio from 'cheerio';
 import generateStats from './svg.js';
 const app = express();
 
-app.get('/', function(req, res) {
+// Get SVG of the User Data
+app.get('/', function (req, res) {
     let userName = req.query.userName;
     if (!userName) {
         let errorMessage = { "error": "add your geeksForGeeks user Name in link eg /?userName=<YOUR_USER_NAME>" };
         res.send(errorMessage);
     } else {
-        let url = "https://auth.geeksforgeeks.org/user/" + userName + "/practice/";
-        request(url, function(error, response, html) {
-            if (!error) {
-                var $ = cheerio.load(html);
-                let values = {};
-                let problemDificultyTag = ["School", "Basic", "Easy", "Medium", "Hard"];
-                let k = 0;
-                // GFG updated UI
-                let data = $('.tabs.tabs-fixed-width.linksTypeProblem');
-
-                if (data.length == 0) {
-                    res.status(400).send({ error: "userName does not exist or not solved any problem on geeksforgeeks" });
-                } else {
-                    let totalProblemSolved = 0;
-
-                    let rawData = $(data[0]).text();
-                    for (let i = 0; i < rawData.length; i++) {
-                        if (rawData[i] == '(') {
-                            let tempStart = i + 1;
-                            while (rawData[i] != ')') {
-                                i++;
-                            }
-                            let tempProblems = parseInt(rawData.substring(tempStart, i));
-                            values[problemDificultyTag[k++]] = tempProblems;
-                            totalProblemSolved += tempProblems;
+        let values = {};
+        fetchGFGUserData(userName, function (err, result) {
+            if (err) {
+                res.send(err);
+            } else {
+                values = result
+                let svg = generateStats(values);
+                res.setHeader("Content-Type", "image/svg+xml");
+                res.setHeader("Cache-Control", "s-max-age=60, stale-while-revalidate");
+                res.send(svg);
+            }
+        });
 
-                        }
-                    }
-                    // old structure of GFG
-                    // data.each((i, x) => {
-                    //     let temp = $(x).text();
-                    //     let number = 0;
-                    //     console.log(temp);
-                    //     for (let i = temp.length - 1; i > 0; i--) {
-                    //         if (temp[i] == ')') {
-                    //             let j = i - 1;
 
-                    //             while (temp[j] != '(') {
+    }
+});
+
 
-                    //                 j--;
-                    //             }
-                    //             number = parseInt(temp.substring(j + 1, i));
-                    //             totalProblemSolved += number;
+// Get Raw Json data of the User
+app.get('/raw', function (req, res) {
```

Comment:
use middleware in chain.
app.get('/', fetchdata,sendRes)
instead of callbacks. you can add validator also, just to explore chain for learning. otherwise as only 1 function, its not needed.